control_server/tests/middleware/generic_message_test.py

```diff
@@ -0,0 +1,77 @@
+from typing import List, Tuple, Dict, Any
+
+import pytest
+
+from control_server.src.middleware.headers.message_header import MessageHeader
+from control_server.src.middleware.messages.generic_message import \
+    GenericMessage
+from control_server.tests.utils.header_utils import HeaderUtils
+
+datasets: List[Tuple[bytes, Dict[str, Any]]] = [
+    (
+        b'\x00\x00\x00\x00\x00\x00\x00\x00\x00AAA',
+        {
+            'message_length': 0,
+            'status_code': 0,
+            'url_length': 0,
+            'body_length': 0,
+            'headers_length': 0,
+            'url': b'A',
+            'body': b'A',
+            'headers': b'A'
+        }
+    ),
+    (
+        b'\x00\x01\x02\x00\x03\x00\x04\x00\x05ABC',
```

Comment:
In this case, it does not matter, because generic_message_test.py only tests that the message is serialized and deserialized correctly, and provides another header in the utils method, that sets URL, body and header length to one. This should however be improved in a future PR to clarify.